ocaml/xapi/vm_platform.ml

```diff
@@ -145,16 +145,22 @@ let sanity_check ~platformdata ?firmware ~vcpu_max ~vcpu_at_startup ~domain_type
   end;
   if check_cores_per_socket && (List.mem_assoc "cores-per-socket" platformdata) then
     begin
+      let cps_str = List.assoc "cores-per-socket" platformdata in
+      let vcpus = Int64.to_int(vcpu_max) in
       try
-        let cores_per_socket = int_of_string(List.assoc "cores-per-socket" platformdata) in
+        let cps = int_of_string(cps_str) in
```

Comment:
Negative integers can be rejected at this stage with Scanf:
```
# Scanf.sscanf "-4" "%u" Fun.id;;
Exception:
Stdlib.Scanf.Scan_failure
 "scanf: bad input at char number 0: character '-' is not a decimal digit".
```

Reply:
Not necessarily a user friendly error though